CONTRIBUTING.md

```diff
@@ -72,17 +73,94 @@ Once merged to the main branch, `po` files and any documentation change will be
 
 ### Required dependencies
 
-TODO
+This project has several build dependencies. You can install these dependencies from the top of the source tree using the `apt` command as follows:
+
+```shell
+sudo apt update
+sudo apt build-dep .
+```
 
 ### Building and running the binaries
 
-TODO
+The project consists of the following binaries:
+
+* `authd`: The main authentication service.
+* `pam_authd.so` and `pam_authd_exec.so`: A PAM module and its helper library.
+* `libnss_authd.so`: An NSS module.
+
+The project can be built as a Debian package. This process will compile all the binaries, run the test suite, and produce the Debian packages.
+
+Alternatively, for development purposes, each binary can be built manually and separately.
+
+#### Building the Debian package from source
+
+Building the Debian package from source is the most straightforward and standard method for compiling the binaries and running the test suite. To do this, run the following command from the top of the source tree:
+
+> [!NOTE]
+> This is required to vendorize the Rust crates and must be only done once.
+> ```shell
+> sudo apt install libssl-dev
+> cargo install cargo-vendor-filterer
+> cargo vendor-filterer vendor_rust
+> ```
+
+Then build the Debian package:
+
+```shell
+dpkg-buildpackage
+```
+
+The debian packages are available in the parent directory.
+
+#### Building authd only
+
+To build `authd` only, run the following command from the top of the source tree:
+
+```shell
+go build ./cmd/authd
+```
+
+The built binary will be in the current directory. The daemon can be run directly from this binary without installing it on the system.
+
+#### Building the PAM module only
+
+To build the PAM module, from the top of the source tree run the following commands:
+
+> [!NOTE]
+> This dependency is required to regenerate the proto files and is only needed once.
+> ```shell
+> sudo apt install protoc-gen-go
```

Comment:
you need to `cd tools/` and go install all the tooling from there too to have the grpc modules in Go hooking up in protoc. I am pretty sure that you don’t need to install the binary package after this and at least, you will be using the version we are pinning in our repo.

Reply:
okay, I'll change with
cd tools/
grep -o '_ ".*"' *.go | cut -d '"' -f 2 | xargs go install

Reply:
done